db/docker-compose.yml

```diff
@@ -10,7 +10,7 @@ services:
         ports:
             - "5438:5432"
         expose:
-            - "5438"
+            - "5432"
```

Comment:
Why this change?

Reply:
`expose` indicates the internal port to be exposed to other containers, and the internal port where the DB is listening is `5432`. So `pgadmin` needs visibility of that port in order to link with the DB.

Reply:
But actually we are exposing 5438 from the container to the outside and forwarding that to PSQL default port 5432 (that's because very often we might run other component's DBs locally so they do not collide), wouldn't it be more correct to setup pgadmin to connect to 5438 port from docker compose file?

Reply:
[ports](https://docs.docker.com/compose/compose-file/compose-file-v2/#ports) and [expose](https://docs.docker.com/compose/compose-file/compose-file-v2/#expose) are different things: `ports` is to define port mappings while `expose` is to expose ports without publishing them to the host machine ( they’ll only be accessible to linked services).

In fact in recent docker versions `expose` is not required anymore, so we could just remove the `expose` entry and connection from `pgadmin` to `vulndb:5432` would work. But I think it's better to keep it just for being explicit.

But `pgadmin` can't connect to `vulndb:5438` because there's no service listening there. If you want to do that or either you configure the postgres to listen `5438` or you might try to connect through the host exposed port (`host.docker.internal:5438`), but what's the point for that?

Reply:
Ok now I got what's going on, was not thinking that _pgadmin_ is a linked container in the same docker compose. Got it! :+1:

Reply:
👍 :)